enhancements/api-review/custom-tags-aws.md

```diff
@@ -31,113 +36,325 @@ have the ability to apply user defined tags to many resources created by OpenShi
+Also refer to `Precedence` scenario where delete of user-defined tag is not allowed to delete.\
+For example,\
+Existing tag for AWS resource = `key_infra1 = value_old`\
+New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 =`\
+Action = Existing tag for AWS resource is deleted.\
+Final tag set to AWS resource = deleted\
+Event action = An event is generated to notify user about the action status (success/failure) to delete tags for the AWS resource.
+
+### Precedence scenarios
+1) User-defined tags on local objects MUST continue to take precedence during create and update. User-defined tags found on local objects must not be deleted by methods described in `Delete tags scenarios`.
+   A warning is generated to inform the user about action not being applied on the list of user tags.
+
+   For example,\
+   Existing tag in local object like machine CRD = `key_infra1 = custom_value`\
+   New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 = value1`\
+   Action = The tag for AWS resource is maintained to `key_infra1 = custom_value`.\
+   Final tag set to AWS resource = `key_infra1 = value_update1`\
+   Event action = An event is generated to notify user about the action status (success/failure) to update tags for the AWS resource. A warning also must be generated about list of user-defined tags on which action is not applicable.
+
+2) `.spec.platformSpec.aws` take precedence over `.status.platformStatus.aws`. User-defined tags must be merged from `.status.platformStatus.aws` and set to AWS resource.
+   On delete, user-defined must also be removed from `.status.platformStatus.aws`, if present.
+
+### Caveats
+1) User updates the user-defined tag from using external tools when there is an entry in `.spec.platformSpec.aws.resourceTags`
+   The user-defined tag which is updated from spec, will be reconciled by operators to set value from `.spec.platformSpec.aws.resourceTags`.
+   The user-defined tag will be overwritten with value from `.spec.platformSpec.aws.resourceTags` when there is an update to `.spec.platformSpec.aws.resourceTags`.
+
+   User must handle inconsistencies in `.spec.platformSpec.aws.resourceTags` and user-defined tag value for AWS resource when using multiple tools to manage tags.
+
+   For example,\
+   Edited existing tag using external tool for AWS resource = `key_infra1 = value_comp1`\
+   Previous tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 = value_infra1`\
+   Action = Update existing tag with value from `.spec.platformSpec.aws.resourceTags`.\
+   Final tag set to AWS resource = `key_infra1 = value_infra1`\
+   Event action = An event is generated to notify user about the action status (success/failure) to update tags for the AWS resource.
+
+2) User deletes the user-defined tag from `.spec.platformSpec.aws.resourceTags`
+   The user-defined tag which is removed from spec, will not be reconciled or managed by operators.\
```

Comment:
This sentence needs to be re-worked. It does not make sense as written. Are you trying to say that an operator is not expected (or even not permitted) to make changes to an AWS resource owned by the operator in response to a user-defined tag being removed from the infrastructure spec?